### PR TITLE
test(spanner): cater to the service vagaries on nested JSON

### PR DESCRIPTION
**Pull request by devbww**
Reduce the "spanner_json_max_nesting_level" from 90 to 80, now that
the service is crashing on nesting depths somewhere in [82..90].
It appears that 80 will be the new documented limit.

And that means we temporarily expect `kAlreadyExists` when trying to
re-add a value with a nesting depth of 81.

See #7212, #8327, and #8689 for previous action in this area.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9727)

### CODE REVIEW

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc [hunk 1]
```diff
@@ -431,8 +431,8 @@ TEST_F(DataTypeIntegrationTest, InsertAndQueryWithStruct) {
 
 // Verify maximum JSON nesting.
 TEST_F(DataTypeIntegrationTest, JsonMaxNesting) {
-  // The default value of the backend max-nesting-level flag.
-  int const k_spanner_json_max_nesting_level = 90;
+  // The default value of the backend spanner_json_max_nesting_level flag.
+  int const k_spanner_json_max_nesting_level = 80;
```
Comment (coryan) [inline google/cloud/spanner/integration_tests/data_types_integration_test.cc:435]: The name looks funny, if this is intended to be a compile-time constant maybe:

```
auto constexpr kSpannerJsonMaxNestingLevel = 80;
```

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc [hunk 2]
```diff
@@ -456,11 +456,21 @@ TEST_F(DataTypeIntegrationTest, JsonMaxNesting) {
     // The emulator has no such limitation, so it tries to re-insert.
     EXPECT_THAT(result, StatusIs(StatusCode::kAlreadyExists));
   } else {
+#if 1
+    // NOTE: The service is currently (August 2022) in a state where there
+    // is no workable "spanner_json_max_nesting_level".  It accepts 80, but
+    // crashes on some yet-to-be discovered value in [82..90].  81 results
+    // in ALREADY_EXISTS.  We really want to get out of the "JsonMaxNesting"
```
Comment (coryan) [inline google/cloud/spanner/integration_tests/data_types_integration_test.cc:463]: Let's do that.  Set the constant to 20.  Or remove this test altogether.  The teams that own this stuff should write their own tests.

**Review by coryan — APPROVED** [review]
I would rather remove this test.  But I guess this is just the second strike.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc:435]: Yeah ... without the `constexpr` that is what's required (I believe).  But I was happy with that as "spanner_json_max_nesting_level" is the true name of the Spanner flag, so it helped when I needed to re-search.

That said, see below.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 2 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc:463]: I'm totally on board with that (removing the test altogether).  But I propose that we wait until the current issue is addressed.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc:435]: You can at least remove the `k_` prefix in that case.

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc:463]: SGTM

File: google/cloud/spanner/integration_tests/data_types_integration_test.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/spanner/integration_tests/data_types_integration_test.cc]: Done.

**Pull request merged**